core/src/main/java/org/apache/iceberg/puffin/PuffinWriter.java

```diff
@@ -144,7 +142,8 @@ public void finish() throws IOException {
     long footerOffset = outputStream.getPos();
     writeFooter();
     this.footerSize = Optional.of(Math.toIntExact(outputStream.getPos() - footerOffset));
-    this.fileSize = Optional.of(outputStream.getPos());
+    outputStream.close();
```

Comment:
seems like the close() can trigger a write to the outputStream and post writer wrote the footer 

https://github.com/apache/iceberg/blob/fb52fdef4fdcaa3cb5779e4d5b1b9f1ea7522e73/core/src/main/java/org/apache/iceberg/encryption/AesGcmOutputStream.java#L162


wondering we should update the description that this patch additionally includes this fix too ?